query.go

```diff
@@ -97,10 +108,42 @@ func getErrorFrameFromQuery(query *Query) data.Frames {
 	return frames
 }
 
+func getQueryFlow(query backend.DataQuery) string {
+	q, _ := GetQuery(query)
+	if q.Meta.QueryFlow == "async" {
+		return "async"
+	}
+	return "sync"
+}
```

Comment:
Helper to check whether the query should be run with the sync flow (current), or the async flow. This is sent from the frontend when it tries to run a query. The frontend reads a feature toggle to determine what to send.

Does it make sense to make these enums or are strings fine here?